src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -36,12 +35,15 @@ function HeatmapVis(props: Props): JSX.Element {
     showGrid = false,
     showLoader = true,
     dataLabel,
+    abscissaParams = {},
+    ordinateParams = {},
   } = props;
 
   const { rows, cols } = getDims(dataArray);
   const aspectRatio = keepAspectRatio ? cols / rows : undefined; // width / height <=> cols / rows
 
-  const { abscissas = range(cols + 1), ordinates = range(rows + 1) } = props;
+  const abscissas = getPixelEdges(abscissaParams.values, cols);
+  const ordinates = getPixelEdges(ordinateParams.values, rows);
 
   if (abscissas.length !== cols + 1) {
     throw new Error(
```

Comment:
These throws can be removed now, right? If `getPixelEdges` throws, then we don't know which axis is concerned, but it's not a big deal. The dimension of the axis is already a good hint.

Reply:
:fishing_pole_and_fish: